seisbench/models/base.py

```diff
@@ -192,6 +193,113 @@ def download_metadata_callback(metadata_path):
+        :type path: pathlib.Path
+        :return: Model instance
+        :rtype: SeisBenchModel
+        """
+        model_files = os.listdir(path)
+
+        if sum([file.endswith("pt") for file in model_files]) > 1:
+            seisbench.logger.warning(
+                f"Strange format of SeisBench model directory detected at {path}. "
+                f"There is more than one .pt file. This will lead to unforseen "
+                f"behaviour during model loading."
+            )
+        if sum([file.endswith("json") for file in model_files]) > 1:
+            seisbench.logger.warning(
+                f"Strange format of SeisBench model directory detected at {path}. "
+                f"There is more than one .json file. This will lead to unforseen "
+                f"behaviour during model loading."
+            )
+
+        for file in model_files:
+            if file.endswith("json"):
+                with open(path / file, "r") as json_file:
+                    weights_metadata = json.load(json_file)
+            elif file.endswith("pt"):
+                model_weights = torch.load(f"{path / file}")
+
+        def _get_args(obj):
+            signature = inspect.signature(obj)
+            args = {k: v._default for k, v in signature.parameters.items()}
+            if "kwargs" in args.keys():
+                del args["kwargs"]
+            return args
+
+        model_args = weights_metadata.get("model_args", {})
+
+        # Overwrite default class args from config model args
+        cls_args = _get_args(cls)
+        cls_args.update({k: model_args[k] for k in cls_args if k in model_args})
+
+        model = cls(**cls_args)
+
+        model._weights_metadata = weights_metadata
+        model._parse_metadata()
+
+        model.load_state_dict(model_weights)
+
+        return model
+
+    def save(self, path=None, name=None):
+        """
+        Save a SeisBench model locally.
+
+        SeisBench models are stored inside the directory 'path'. SeisBench models are saved in 2 parts,
+        the model configuration is stored in JSON format [path]/[name.json], and the underlying model weights
+        in PyTorch format [path]/[name].pt.
+
+        The model config should contain the following information, which is automatically created from
+        the model instance state:
+            - "docstring": A string documenting the pipeline. Usually also contains information on the author.
+            - "model_args": Argument dictionary passed to the init function of the pipeline.
+            - "seisbench_requirement": The minimal version of SeisBench required to use the weights file.
+            - "default_args": Default args for the :py:func:`annotate`/:py:func:`classify` functions.
+
+        Non-serlisable arguments (e.g. functions) cannot be saved to JSON, so are not converted.
+
+        :param path: Define the output model directory. If none is provided, the path will create a directory
```

Comment:
I'd not create the directory if missing but throw an exception, just as the python `write` function.